### PR TITLE
Fix packages tests

### PR DESCRIPTION
**Pull request by fmanno**
[Linux/macOS] Fix data races:
Move the denitialization of the watcher to when
the fsEvents channel is stopped.

[all] Fix error handling of TestOpenProject:
If the result of w.OpenProject was nil. The test
was exploding

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -29,8 +29,12 @@ addons:
       - xclip
 
 install:
+  - cd $HOME
+  - mkdir -p $HOME/gopath/src/github.com/limetext
+  - mv ${TRAVIS_BUILD_DIR} $HOME/gopath/src/github.com/limetext/backend
+  - cd $HOME/gopath/src/github.com/limetext/backend
+  - export TRAVIS_BUILD_DIR="$HOME/gopath/src/github.com/limetext/backend"
```
Comment (erbridge) [inline .travis.yml]: Could you revert these changes, please?

Comment (fmanno) [inline .travis.yml]: Sorry, my bad. I was testing on a fork and I forgot to remove this before doing the pull request. I'll fix it later today.

**Comment by zoli** [comment]
I have an idea why the test still fails for `TestWatchDir`, will continue from your changes. Thanks
